src/js/helpers.js

```diff
@@ -1130,4 +1262,169 @@ export default class Helpers {
+        result['columnInfo'] = parseResult
+        cleanedFields.push(result)
+      }
+    }
+
+    return cleanedFields
+  }
+
+  tryParseColumnInfo(data) {
+    const result = {}
+
+    if (data.className) {
+      const classes = data.className
+        .split(' ')
+        .filter(className => /^col-(xs|sm|md|lg)-([^\s]+)/.test(className) || className.startsWith('row-'))
+
+      if (classes && classes.length > 0) {
+        classes.forEach(element => {
+          if (element.startsWith('row-')) {
+            result['rowNumber'] = parseInt(element.replace('row-', '').trim())
+          } else {
+            result['columnSize'] = element
+          }
+        })
+      }
+    }
+
+    return result
+  }
+
+  toggleGridModeButtonVisible(fieldId, visible = true) {
+    if (!visible) {
+      $(`#${fieldId}-grid`).css('display', 'none')
+      return
+    }
+
+    $(`#${fieldId}-grid`).css('display', '')
+  }
+
+  //Briefly highlight on/off
+  toggleHighlight(field, ms = 600) {
+    field.addClass('moveHighlight')
+    setTimeout(function () {
+      field.removeClass('moveHighlight')
+    }, ms)
+  }
+
+  showToast(msg, timeout = 3000) {
+    if (this.toastTimer != null) {
+      window.clearTimeout(this.toastTimer)
+      this.toastTimer = null
+    }
+
+    this.toastTimer = setTimeout(function () {
+      $('.snackbar').removeClass('show')
+    }, timeout)
+
+    $('.snackbar').addClass('show').html(msg)
+  }
+
+  getDistanceBetweenPoints(x1, y1, x2, y2) {
+    const y = x2 - x1
+    const x = y2 - y1
+
+    return Math.floor(Math.sqrt(x * x + y * y))
+  }
+
+  //Return full row name (row-1)
+  getRowClass(className) {
+    if (!className) {
+      return
+    }
+
+    const splitClasses = className.split(' ').filter(x => x.startsWith('row-'))
+    if (splitClasses && splitClasses.length > 0) {
+      return splitClasses[0]
+    }
+  }
+
+  //Return the row value i.e row-2 would return 2
+  getRowValue(className) {
+    if (!className) {
+      return 0
+    }
+
+    const rowClass = this.getRowClass(className)
+    if (rowClass) {
+      return parseInt(rowClass.split('-')[1])
+    }
+  }
+
+  //Return the column size i.e col-md-6 would return 6
+  getBootstrapColumnValue(className) {
+    if (!className) {
+      return 0
+    }
+
+    const bootstrapClass = this.getBootstrapColumnClass(className)
+    if (bootstrapClass) {
+      return parseInt(bootstrapClass.split('-')[2])
+    }
+  }
+
+  //Return the prefix (col-md)
+  getBootstrapColumnPrefix(className) {
+    if (!className) {
+      return 0
+    }
+
+    const bootstrapClass = this.getBootstrapColumnClass(className)
+    if (bootstrapClass) {
+      return `${bootstrapClass.split('-')[0]}-${bootstrapClass.split('-')[1]}`
+    }
+  }
+
+  //Return full class name (col-md-6)
+  getBootstrapColumnClass(className) {
+    if (!className) {
+      return
+    }
+
+    const splitClasses = className.split(' ').filter(className => /^col-(xs|sm|md|lg)-([^\s]+)/.test(className))
```

Comment:
`/^col-(xs|sm|md|lg)-([^\s]+)/` regexp is used in a few places, maybe we should move it to utils.js so it can be imported where needed

Reply:
I like it. I also refactored some of the other surrounding usages